hadoop-tools/hadoop-sls/src/test/java/org/apache/hadoop/yarn/sls/TestSLSDagAMSimulator.java

```diff
@@ -62,6 +66,8 @@ public static Collection<Object[]> data() {
 
   @Before
   public void setup() {
+    //JavaScript engine has been removed from Java in Java 15.
+    assumeNotNull(new ScriptEngineManager().getEngineByName("JavaScript"));
```

Comment:
this is a common assert. why not pull it out into some shared utility class

+ add a text message as to why it was skipped, just for those test reports